extern/storage-sealing/states_replica_update.go

```diff
@@ -203,9 +206,87 @@ func (m *Sealing) handleReplicaUpdateWait(ctx statemachine.Context, sector Secto
 }
 
 func (m *Sealing) handleFinalizeReplicaUpdate(ctx statemachine.Context, sector SectorInfo) error {
+	cfg, err := m.getConfig()
+	if err != nil {
+		return xerrors.Errorf("getting sealing config: %w", err)
+	}
+
+	if err := m.sealer.FinalizeReplicaUpdate(sector.sealingCtx(ctx.Context()), m.minerSector(sector.SectorType, sector.SectorNumber), sector.keepUnsealedRanges(false, cfg.AlwaysKeepUnsealedCopy)); err != nil {
+		return ctx.Send(SectorFinalizeFailed{xerrors.Errorf("finalize sector: %w", err)})
+	}
+
 	return ctx.Send(SectorFinalized{})
 }
 
+func (m *Sealing) handleUpdateActivating(ctx statemachine.Context, sector SectorInfo) error {
+	try := func() error {
+		mw, err := m.Api.StateWaitMsg(ctx.Context(), *sector.ReplicaUpdateMessage)
+		if err != nil {
+			return err
+		}
+
+		tok, height, err := m.Api.ChainHead(ctx.Context())
+		if err != nil {
+			return err
+		}
+
+		nv, err := m.Api.StateNetworkVersion(ctx.Context(), tok)
+		if err != nil {
+			return err
+		}
+
+		lb := policy.GetWinningPoStSectorSetLookback(nv)
+
+		targetHeight := mw.Height + lb + InteractivePoRepConfidence
+		delay := 50 * time.Millisecond
+
+		for {
+			if height >= targetHeight {
+				break
+			}
+
+			wctx, cancel := context.WithTimeout(ctx.Context(), delay)
+			<-wctx.Done()
+			cancel()
+
+			// increasing backoff; can't just calculate the correct time because integration tests do funny things with time
+			delay = delay * 10 / 3
+			if delay > 5*time.Minute {
```

Comment:
What's the benefit of capping at 5 minutes?  Isn't the winning post lookup about a day long?

Reply:
Hrm, there is a better way